openshift/system/config/amazon_s3.yml

```diff
@@ -8,6 +8,7 @@ s3: &s3
   region:            "<%= ENV['AWS_REGION'] %>"
   hostname:          "<%= ENV['AWS_HOSTNAME'] %>"
   protocol:          "<%= ENV['AWS_PROTOCOL'] %>"
+  force_path_style:  <%= ENV['AWS_PATH_STYLE'].presence || false %>
```

Comment:
@eguzki , I'm afraid we'll need to introduce one more env var, `AWS_PATH_STYLE`.

We cannot enforce it by default because path-style endpoints will [lose support on 30 Sep 2020](https://aws.amazon.com/blogs/aws/amazon-s3-path-deprecation-plan-the-rest-of-the-story/). Other "API-compatible" services like minio.io, however, may still require it (at least, http://play.minio.io does).

Reply:
If we decide not to implement this new env var, a workaround is including a dot in the name of the bucket. This will enforce path-style endpoints despite the value of the `force_path_style` configuration option. (Only valid when the protocol is `https`.)

https://github.com/aws/aws-sdk-ruby/blob/9409865777f15c7db2a8b6937fdd571082d12cce/aws-sdk-resources/lib/aws-sdk-resources/services/s3/bucket.rb#L111

Reply:
AWS_PATH_STYLE will be there

Reply:
PR with the new env var: https://github.com/3scale/3scale-operator/pull/306

Reply:
I am not sure to understand correctly this `force_path_style` configuration.

Why do we need it? I can't see any discussion in the issue or its comments about this
I just want to understand the need of this, or lets document what is missing in the issue (for future archeologist)

Reply:
This is because https://github.com/aws/aws-sdk-ruby tests if the name of the bucket is "dns-compatible". Whenever it sees a bucket named, for example, "mybucket", it will build an endpoint using the subdomain style: i.e. "https://mybucket.host". On the other hand, if the bucket is named "my.bucket" (i.e. not "dns-compatible"), the endpoint will be built using path style: i.e. "https://host/my.bucket".

http://play.minio.io requires path style always, no matter if the bucket name is "dns compatible" or not. Not having the option to set `force_path_style` would make 3scale incompatible with use cases where the name of the bucket is "dns-compatible" but the endpoint must still be in path-style.